test/distributed/_sharded_tensor/ops/test_linear.py

```diff
@@ -44,24 +48,32 @@
 
 
 class TestShardedTensorOpsLinear(ShardedTensorTestBase):
-    def _run_sharded_linear(self, spec, input_size, linear_size, sharded_dim):
+    def _run_sharded_linear(
+        self, spec, input_size, linear_size, sharded_dim, unstack_output=False,
+    ):
         # Use same seed.
         torch.manual_seed(0)
         local_linear = torch.nn.Linear(*linear_size).cuda(self.rank)
-
         sharded_linear = torch.nn.Linear(*linear_size)
 
         # Copy the weights and bias from local linear
-        sharded_linear.weight = torch.nn.Parameter(local_linear.weight.detach().clone())
-        sharded_linear.bias = torch.nn.Parameter(local_linear.bias.detach().clone())
+        sharded_linear.weight = clone_module_parameter(local_linear, "weight")
+        sharded_linear.bias = clone_module_parameter(local_linear, "bias")
 
         # Shard the parameter.
         shard_parameter(sharded_linear, "weight", spec)
 
         # Run sharded computation
         torch.manual_seed(self.rank)  # inputs different on each rank
         inp = torch.rand(*input_size).cuda(self.rank)
+        reshard_spec = copy.deepcopy(spec)
+        reshard_spec.dim = 0
+        sharded_linear = _collect_local_shard(
+            _reshard_output(sharded_linear, reshard_spec)
+        )
         sharded_output = sharded_linear(inp)
+        if unstack_output:
+            sharded_output = torch.cat(torch.unbind(sharded_output))
```

Comment:
Why do we need to do this? Try to write unit tests from the users point of view to see if the APIs we introduced are user friendly or not.

For example, the idea of adding `_collect_local_shard` and `_reshard_output` was to ensure the user doesn't have to do additional operations like this to get the desired result.

Reply:
This is because we have this tricky corner case when the input is only one dimension. The col-wise sharding will concatenate all local results also in the dimension 0. I tried a workaround to make it `stack` instead, that's why we need an `unbind` here. Let me think more on this to see if we can have a cleaner solution.

Reply:
Another work-around is that for this type of case, we will make the shard tensor shard by dim -1. So that we can set the dim back to the last dim during the reshard work. (Because by default when sharding_dim = = reshard_dim, for this case both == 0, we just reshuffle the local shard). What do you think?